### DISCUSSION TITLE
Support use path aliases to import fonts and other files

### DISCUSSION THREAD

**Discussion by mcitem** [slint-ui/slint#7085] [Ideas]
![image](https://github.com/user-attachments/assets/69e12073-5cc3-4d28-bac6-1a77486894be)
I think using relative paths will eventually turn into a disaster.
I hope to import other resource files (such as fonts) through aliases just like importing Slint, similar to the path aliases in Vite.
https://vite.dev/config/shared-options.html#resolve-alias

**Comment by ogoffart** [comment]
Looks like a bug to me. Created https://github.com/slint-ui/slint/issues/7086

**Comment by ogoffart** [comment]
Btw, i believe what can be done is just to import these font from a .slint file in that library and then they should be imported.

**Reply by mcitem** [reply]
> Btw, i believe what can be done is just to import these font from a .slint file in that library and then they should be imported.

Perhaps I abused the library import feature ();
After all, the difference between a library import and a path alias is quite significant.
```json
{
  "slint.libraryPaths": {
    "ui": "ui\\",
    "a": "ui\\assets\\",
    "views": "ui\\views\\",
    "c": "ui\\components\\",
    "sui": "ui\\modules\\surrealism-ui\\"
  }
}
```
```rs
fn main() {
    let manifest_dir = std::path::PathBuf::from(std::env::var_os("CARGO_MANIFEST_DIR").unwrap());
    let library_paths = std::collections::HashMap::from([
        ("ui".into(), manifest_dir.join("ui")),
        ("a".into(), manifest_dir.join("ui/assets")),
        ("views".into(), manifest_dir.join("ui/views")),
        ("c".into(), manifest_dir.join("ui/components")),
        ("sui".into(), manifest_dir.join("ui/modules/surrealism-ui")),
    ]);
    let config = slint_build::CompilerConfiguration::new().with_library_paths(library_paths);
    slint_build::compile_with_config("ui/app.slint", config).unwrap();
}

```
